beacon-chain/sync/validate_aggregate_proof.go

```diff
@@ -131,21 +132,34 @@ func (s *Service) validateAggregatedAtt(ctx context.Context, signed *ethpb.Signe
 		return pubsub.ValidationReject
 	}
 
-	// Verify selection proof reflects to the right validator and signature is valid.
-	if err := validateSelection(ctx, bs, signed.Message.Aggregate.Data, signed.Message.AggregatorIndex, signed.Message.SelectionProof); err != nil {
+	// Verify selection proof reflects to the right validator.
+	selectionSigSet, err := validateSelectionIndex(ctx, bs, signed.Message.Aggregate.Data, signed.Message.AggregatorIndex, signed.Message.SelectionProof)
+	if err != nil {
 		traceutil.AnnotateError(span, errors.Wrapf(err, "Could not validate selection for validator %d", signed.Message.AggregatorIndex))
-		return pubsub.ValidationReject
+		return pubsub.ValidationIgnore
 	}
 
-	// Verify the aggregator's signature is valid.
-	if err := validateAggregatorSignature(bs, signed); err != nil {
+	// Verify selection signature, aggregator signature and attestation signature are valid.
+	// We use batch verify here to save compute.
+	aggregatorSigSet, err := aggSigSet(bs, signed)
+	if err != nil {
+		traceutil.AnnotateError(span, errors.Wrapf(err, "Could not get aggregator sig set %d", signed.Message.AggregatorIndex))
+		return pubsub.ValidationIgnore
+	}
+	attSigSet, err := blocks.AttestationSignatureSet(ctx, bs, []*ethpb.Attestation{signed.Message.Aggregate})
+	if err != nil {
 		traceutil.AnnotateError(span, errors.Wrapf(err, "Could not verify aggregator signature %d", signed.Message.AggregatorIndex))
-		return pubsub.ValidationReject
+		return pubsub.ValidationIgnore
```

Comment:
same here

Reply:
Failing to construct attestation signature set is a self-error and we shouldn't stop broadcasting it to our peers

Reply:
That depends on what kind of error, ex:
https://github.com/prysmaticlabs/prysm/blob/86e4c90fd999d9045e60cb76d2284d86ba0081da/beacon-chain/core/blocks/signature.go#L127:6

Most of the errors there would only be possible because of an invalid attestation aggregate. Ignoring rather than rejecting something like this seems risky.

Reply:
Not necessarily. These errors could come from:
 * Bad implementation of caching (e.g. BeaconCommitteeFromState)
 * Bad implementation of hash tree root (e.g. ComputeSigningRoot)
 * Bad implementation of helper functions (e.g. ConvertToIndexed, IsValidAttestationIndices)

These are likely client independent bug that varies from releases and languages. Example, if prysm has a bug on a specific version, it'll kill liveness for lh and nimbus as well. We don't want to stop broadcast and descore peers due to reasons above.

Likewise, if the aggregation is really bad, the signature validation will fail which will cause a reject. It feels safer on the liveness to be lenient on the first check